streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/ClientState.java

```diff
@@ -151,6 +158,35 @@ else if (thisLoad > otherLoad)
             return capacity > other.capacity;
     }
 
+    boolean hasMoreAvailableCapacityThan(final ClientState other) {
+        checkCapacity(other);
+
+        final double otherLoad = (double) other.assignedTaskCount() / other.capacity();
+        final double thisLoad = (double) assignedTaskCount() / capacity();
+
+        return leastLoaded(thisLoad, otherLoad, other);
+    }
+
+    boolean hasMoreAvailableActiveTaskCapacityThan(final ClientState other) {
```

Comment:
nit: method name `hasMoreAvailableActiveTaskCapacityThan`  -> `hasMoreAvailableActiveStoreTaskCapacityThan` or something along those lines, I found this method name a little confusing with `hasMoreAvailableCapacityThan`.